### PR TITLE
feat: add slot stats to /agents endpoints

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

waiting on web to confirm the schema is sufficient

## Test Plan
a single   map<string, DeviceStats> type_stats = 6; seems enough for cluster ux in its current state 
the rest are just artifacts of incremental change requests

## Commentary (optional)

putting this up to get early feedback on the schema changes

not going to land any of the web changes via this pr

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

related to https://hpe-aiatscale.atlassian.net/browse/RM-108

### CODE REVIEW

File: proto/src/determined/agent/v1/agent.proto [hunk 1]
```diff
@@ -9,6 +9,48 @@ import "protoc-gen-swagger/options/annotations.proto";
...
+  // The number of slots that are draining.
+  int32 draining_count = 3;
+  // Map of slot ids to container state, if there is an associated container.
+  map<string, determined.container.v1.State> slot_states = 4;
+  // Set of slot ids that are disabled.
+  repeated string disabled_slots = 5;
+  // Map of device type to device stats.
+  map<string, DeviceStats> type_stats = 6;
+  // Map of device brands to device stats.
+  map<string, DeviceStats> brand_stats = 7;
```
Comment (hamidzr) [inline proto/src/determined/agent/v1/agent.proto:51]: just keep these two

File: proto/src/determined/agent/v1/agent.proto [hunk 2]
```diff
@@ -36,6 +78,30 @@ message Agent {
   // The name of the resource pools the agent is in. Only slurm can contain
   // multiples.
   repeated string resource_pools = 6;
+
+  // The slot stats for this agent.
+  SlotStats slot_stats = 11;
```
Comment (keita-determined) [inline proto/src/determined/agent/v1/agent.proto:83]: https://github.com/determined-ai/determined/pull/9056#discussion_r1540115545

this should be in required field?

File: webui/react/src/pages/ResourcePool/Topology.tsx [hunk 3]
```diff
@@ -59,6 +59,19 @@ const NodeElement: React.FC<PropsWithChildren<NodeElementProps>> = ({ name, slot
 };
 
 const Topology: React.FC<PropsWithChildren<Props>> = ({ nodes }) => {
+  const MAX_USABLE_NODES = 100;
+  if (nodes.length > MAX_USABLE_NODES) {
+    return (
+      <Section title="Topology">
+        <div className={css.warningContainer}>
+          <p>
+            Current Topology view is disabled for pools with more than {MAX_USABLE_NODES} nodes.
```
Comment (kkunapuli) [inline webui/react/src/pages/ResourcePool/Topology.tsx:68]: Nice!

File: proto/src/determined/agent/v1/agent.proto (hunk 1 above)
Comment (kkunapuli) [reply proto/src/determined/agent/v1/agent.proto:51]: I don't understand?

**Review by kkunapuli — APPROVED** [review]

File: proto/src/determined/agent/v1/agent.proto (hunk 1 above)
Comment (hamidzr) [reply proto/src/determined/agent/v1/agent.proto:51]: reminder for myself. my plan is to just these keep these two new structs and we shouldn't need the rest.

**Comment by keita-determined** [comment]
changed title with `fix` prefix because this PR and web PR should be tested in release party

**Comment by hamidzr** [comment]
> changed title with `fix` prefix because this PR and web PR should be tested in release party

I'd change it to feat: that'd also be tested. I think that'd be more accurate.

**Pull request merged**